services/bls_aggregation/blsagg.go

```diff
@@ -0,0 +1,387 @@
+					SignersAggSigG1:              digestAggregatedOperators.signersAggSigG1,
+					NonSignerQuorumBitmapIndices: indices.NonSignerQuorumBitmapIndices,
+					QuorumApkIndices:             indices.QuorumApkIndices,
+					TotalStakeIndices:            indices.TotalStakeIndices,
+					NonSignerStakeIndices:        indices.NonSignerStakeIndices,
+				}
+				a.aggregatedResponsesC <- blsAggregationServiceResponse
+				return
+			}
+		case <-taskExpiredTimer.C:
+			a.aggregatedResponsesC <- BlsAggregationServiceResponse{
+				Err: TaskExpiredError,
+			}
+			return
+		}
+	}
+
+}
+
+// closeTaskGoroutine is run when the goroutine processing taskIndex's task responses ends (for whatever reason)
+// it deletes the response channel for taskIndex from a.taskChans
+// so that the main thread knows that this task goroutine is no longer running
+// and doesn't try to send new signatures to it
+func (a *BlsAggregatorService) closeTaskGoroutine(taskIndex types.TaskIndex) {
+	a.taskChansMutex.Lock()
+	delete(a.signedTaskRespsCs, taskIndex)
+	a.taskChansMutex.Unlock()
+}
+
+// verifySignature verifies that a signature is valid against the operator pubkey stored in the
+// operatorsAvsStateDict for that particular task
+// TODO(samlaf): right now we are only checking that the *digest* is signed correctly!!
+// we could be sent a signature of any kind of garbage and we would happily aggregate it
+// this forces the avs code to verify that the digest is indeed the digest of a valid taskResponse
+// we could take taskResponse as an interface{} and have avs code pass us a taskResponseHashFunction
+// that we could use to hash and verify the taskResponse itself
+func (a *BlsAggregatorService) verifySignature(
+	taskIndex types.TaskIndex,
+	signedTaskResponseDigest types.SignedTaskResponseDigest,
+	operatorsAvsStateDict map[types.OperatorId]types.OperatorAvsState,
+) error {
+	_, ok := operatorsAvsStateDict[signedTaskResponseDigest.OperatorId]
+	if !ok {
+		a.logger.Warnf("Operator %#v not found. Skipping message.", signedTaskResponseDigest.OperatorId)
+		return OperatorNotPartOfTaskQuorumErrorF(signedTaskResponseDigest.OperatorId, taskIndex)
+	}
+
+	// 0. verify that the msg actually came from the correct operator
+	operatorG2Pubkey := operatorsAvsStateDict[signedTaskResponseDigest.OperatorId].Pubkeys.G2Pubkey
+	if operatorG2Pubkey == nil {
+		a.logger.Fatalf("Operator G2 pubkey not found")
+	}
+	a.logger.Debug("Verifying signed task response digest signature",
+		"operatorG2Pubkey", operatorG2Pubkey,
+		"taskResponseDigest", signedTaskResponseDigest.TaskResponseDigest,
+		"blsSignature", signedTaskResponseDigest.BlsSignature,
+	)
+	signatureVerified, err := signedTaskResponseDigest.BlsSignature.Verify(operatorG2Pubkey, signedTaskResponseDigest.TaskResponseDigest)
+	if err != nil {
+		a.logger.Errorf(SignatureVerificationError(err).Error())
+		return SignatureVerificationError(err)
+	}
+	if !signatureVerified {
+		a.logger.Errorf(IncorrectSignatureError.Error())
```

Comment:
can just use `Error` if not use string formatting

Reply:
[3a8dd29](https://github.com/Layr-Labs/eigensdk-go/pull/24/commits/3a8dd2957ca7dfe8694773728008e7cca5eb60c1)